### PR TITLE
IE11 fixes, dir and theme improvements

### PR DESCRIPTION
**Pull request by paulcpederson**
This builds off @macandcheese 's draft ie11 pr and solves several issues:

1. Almost all of the colors from light and dark theme work in IE11
2. No longer need to set the `dir` programmatically with `getElementDir`
3. No longer need to set a theme to get the correct colors (ie. if theme is not declared it will look like light, even without the attribute)
4. Direction is now inherited, so if you use `dir="rtl"` on the html element, everything will inherit the right to left direction automatically without recomputing anything
5. Theme is now inherited. This means if you don't declare a theme on on the children they will inherit the parent's theme. This is done without any dom traversal or javascript.
6. No longer declaring all the variables in every component every time (one global set of variables)
7. Entire app can be themed very easily
8. Updated tests to stop checking for theme default and validation as it is no longer needed.

### Theming

Now, you can write a simple theme and have those values inherited throughout your app:

```css
.my-cool-theme * {
    --calcite-ui-blue: purple;
}
```

```html
<body class="my-cool-theme"></body>
```

All components in the body would then have purple instead of blue ✨ 

### Right-to-left

Now you can easily swap all the components inside a dom tree back and forth between rtl and ltr:

```
<div class="js-container">
  <calcite-card></calcite-card>
  <calcite-dropdown></calcitedropdown>
  ...
</div>
<script>
 document.querySelector(".js-container").dir = "rtl";
</script>
```

All the enclosed components will be rtl now, no watchers or listeners or dom traversal is needed.

### CODE REVIEW

**Comment by jbartley** [comment]
That's awesome!!

**Comment by paulcpederson** [comment]
Now that the components will load on an ie11 page we can go through and figure out the issues on a case-by-case basis. I think the main problems I saw were:

- icon component doesn't work (path data fails on request) as ie doesn't understand export syntax
- lots of small spacing issues (probably flexbox related)
- some color inheritance issues remain (very few)

**Comment by driskull** [comment]
💯

**Comment by paulcpederson** [comment]
Also @kat10140 there are a couple tests for the card component that are broken, but I couldn't figure out how. Not sure if they were broken on master or if there is some weird edge case, but something to be aware of.

File: src/components/calcite-dropdown-item/calcite-dropdown-item.scss [hunk 1]
```diff
@@ -50,7 +50,7 @@
   opacity: 1;
 }
 
-:host([dir="rtl"]) {
+:host-context([dir="rtl"]) {
```
Comment (driskull) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.scss]: Awesome 👏

**Comment by paulcpederson** [comment]
rtl example 😄 🎥 

1-capture

**Comment by julio8a** [comment]
@macandcheese and @paulcpederson amazing work! you guys are 🧙‍♂️'s
